### PR TITLE
Correctly detect relationship type for form generation

### PR DESCRIPTION
**Pull request by jrjohnson**
Also added required => false flag to field types that are nullable or boolean so HTML5 validator doesn't jump in.

Oh and fixed a massive typo in the service definition.

### CODE REVIEW

File: Resources/skeleton/form/FormType.php.twig [hunk 1]
```diff
@@ -26,13 +26,17 @@ class {{ entity_class }}Type extends AbstractType
 
 {% endif %}
 {% if rest_support and field.fieldName in associations|keys %}
-    ->add('{{ field.fieldName }}', 'tdn_single_related', [
-        'required' => false,
-        'entityName' => "{{ field.targetEntity|addslashes }}"
-    ])
+            ->add('{{ field.fieldName }}', 'tdn_{{ field.realtedType }}_related', [
```
Comment (vpassapera) [inline Resources/skeleton/form/FormType.php.twig]: realtedType -> relatedType

File: Resources/skeleton/form/FormType.php.twig [hunk 2]
```diff
@@ -26,13 +26,17 @@ class {{ entity_class }}Type extends AbstractType
...
+                'required' => false,
+                'entityName' => "{{ field.relatedEntityShrotcut }}"
```
Comment (vpassapera) [inline Resources/skeleton/form/FormType.php.twig]: relatedEntityShrotcut -> relatedEntityShortcut

**Comment by vpassapera** [comment]
This looks great. Minor changes to some var names, but other than that, looking forward to merge!

Ty!

:thumbsup:

**Comment by jrjohnson** [comment]
Wow.  Nice catch(es).  I spelled related wrong in some other code last week too.  Apparently second grade really was important.

**Pull request merged**